### PR TITLE
bugfix: might get previous nextupdate value

### PR DESCRIPTION
**Pull request by ssdr**
For the ocsp response without nextUpdate value, it will get the previous nextUpdate value which is cached in the module-level lua ffi variable, this is not what we expected.

Maybe we should use function-level local variable instead?

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

### CODE REVIEW

**Comment by ssdr** [comment]
https://github.com/openresty/lua-resty-core/commit/7a6521deb7a354fac9cb24cb477c144559ec3c06

**Comment by Appla** [comment]
How about resetting next_update_p by `next_update_p[0] = 0` when `next_update != 0` before return next_update?

**Comment by ssdr** [comment]
> How about resetting next_update_p by `next_update_p[0] = 0` when `next_update != 0` before return next_update?

yes, it seems better.

File: lib/ngx/ocsp.lua [hunk 1]
```diff
@@ -110,7 +110,7 @@ function _M.validate_ocsp_response(resp, chain, max_errmsg_len)
 
     local sizep = get_size_ptr()
     sizep[0] = errbuf_size
-
+
```
Comment (zhuizhuhaomeng) [inline lib/ngx/ocsp.lua:113]: should remove the extra space.

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: lib/ngx/ocsp.lua (hunk 1 above)
Comment (ssdr) [reply lib/ngx/ocsp.lua:113]: accept.

File: lib/ngx/ocsp.lua [hunk 2]
```diff
@@ -119,7 +119,9 @@ function _M.validate_ocsp_response(resp, chain, max_errmsg_len)
     if rc == FFI_OK then
         local next_update = tonumber(next_update_p[0])
         if next_update == 0 then
-          next_update = nil
+            next_update = nil
+        else
+            next_update_p[0] = 0
```
Comment (zhuizhuhaomeng) [inline lib/ngx/ocsp.lua:124]: Would you please put this line before line 114?

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: lib/ngx/ocsp.lua (hunk 2 above)
Comment (ssdr) [reply lib/ngx/ocsp.lua:124]: sure.

**Review by zhuizhuhaomeng — APPROVED** [review]

**Pull request merged**